pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -54,14 +54,15 @@ declare -gi depCacheInitialised=0
 declare -gi doneRecursiveSearch=0
 declare -g foundDependency
 
-getDepsFromSo() {
-    ldd "$1" 2> /dev/null | sed -n -e 's/[^=]*=> *\(.\+\) \+([^)]*)$/\1/p'
+getDepsFromElfBinary() {
+    $READELF -d "$1" 2> /dev/null | \
+        sed -n -e 's/^[\t ]*0x[0-9a-f]\{1,\}[\t ]\{1,\}(NEEDED)[\t ]\{1,\}Shared library: \[\([^]]\{1,\}\)\]/\1/p'
 }
 
 populateCacheWithRecursiveDeps() {
     local so found foundso
     for so in "${autoPatchelfCachedDeps[@]}"; do
-        for found in $(getDepsFromSo "$so"); do
+        for found in $(getDepsFromElfBinary "$so"); do
```

Comment:
I don't think this change is doing the right thing because the library path isn't evaluated yet. I'll take another pass at this.

Reply:
OK, I think I've addressed both of these now. Let me know what you think!